### PR TITLE
Add near prometheus exporter

### PR DESCRIPTION
**Pull request by yanganto**

### CODE REVIEW

File: nix/modules/near-prometheus-exporter.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{ config, lib, ... }:
+let
+  cfg = config.kuutamo.exporter or { accountId = ""; externalRpc = ""; };
+in
+{
+  options.kuutamo.exporter = {
+    accountId = lib.mkOption {
+      type = lib.types.str;
```
Comment (Mic92) [inline nix/modules/near-prometheus-exporter.nix:8]:
```suggestion
      type = lib.types.nullOr lib.types.str;
      default = null;
```

File: nix/modules/near-prometheus-exporter.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    };
+    package = lib.mkOption {
+      type = lib.types.package;
+      defaultText = lib.literalExpression "pkgs.near-prometheus-exporter";
+      description = "The near prometheus exporter package to use in our service";
+    };
+  };
+  config = {
+    systemd.services.near-prometheus-exporter = {
+      enable = if config.kuutamo.exporter.accountId == "" then false else true;
```
Comment (Mic92) [inline nix/modules/near-prometheus-exporter.nix:23]:
```suggestion
```

File: nix/hosts/flake-module.nix [hunk 3]
```diff
@@ -22,6 +22,11 @@
             users.extraUsers.root.openssh.authorizedKeys.keys = [
               "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKbBp2dH2X3dcU1zh+xW3ZsdYROKpJd3n13ssOP092qE joerg@turingmachine"
             ];
+
+            kuutamo.exporter = {
+              accountId = lib.mkDefault "";
+              externalRpc = lib.mkDefault "";
+            };
           };
```
Comment (Mic92) [inline nix/hosts/flake-module.nix:30]:
```suggestion
```

File: nix/hosts/flake-module.nix [hunk 4]
```diff
@@ -22,6 +22,11 @@
             users.extraUsers.root.openssh.authorizedKeys.keys = [
               "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKbBp2dH2X3dcU1zh+xW3ZsdYROKpJd3n13ssOP092qE joerg@turingmachine"
             ];
+
+            kuutamo.exporter = {
+              accountId = lib.mkDefault "";
+              externalRpc = lib.mkDefault "";
+            };
```
Comment (Mic92) [inline nix/hosts/flake-module.nix:29]:
```suggestion
```

File: nix/modules/toml-mapping.nix [hunk 5]
```diff
@@ -35,5 +35,6 @@ in
 
     kuutamo.telegraf.hasMonitoring = cfg.telegraf_has_monitoring or false;
     kuutamo.telegraf.configHash = cfg.telegraf_config_hash or "";
+    kuutamo.exporter.accountId = cfg.validator_account_id or "";
```
Comment (Mic92) [inline nix/modules/toml-mapping.nix:38]:
```suggestion
    kuutamo.exporter.accountId = cfg.validator_account_id or null;
```

File: nix/modules/near-prometheus-exporter.nix (hunk 1 above)
Comment (Mic92) [reply nix/modules/near-prometheus-exporter.nix:8]: We commonly use null to indicate an unset value.

File: nix/modules/near-prometheus-exporter.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      description = lib.mdDoc ''public rpc source'';
+    };
+    package = lib.mkOption {
+      type = lib.types.package;
+      defaultText = lib.literalExpression ".#near-prometheus-exporter";
+      description = lib.mdDoc "The near prometheus exporter package to use in our service";
+    };
+  };
+  config = {
+    systemd.services.near-prometheus-exporter = lib.mkIf (cfg.accountId != null) {
```
Comment (Mic92) [inline nix/modules/near-prometheus-exporter.nix:23]:
```suggestion
  config = lib.mkIf (cfg.accountId != null) {
    services.telegraf.extraConfig.inputs.prometheus.urls = [
      "http://localhost:9333"
    ];

    systemd.services.near-prometheus-exporter = {
```

File: nix/modules/near-prometheus-exporter.nix [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Restart = "always";
+        ExecStart = ''${cfg.package}/bin/near-exporter \
+          -accountId ${cfg.accountId} \
+          -addr [::1]:9333 \
```
Comment (Mic92) [inline nix/modules/near-prometheus-exporter.nix:29]: Could you also test for localhost:9333 in the neard nixos test to see if the prometheus exporter is started correctly?

Comment (Mic92) [reply nix/modules/near-prometheus-exporter.nix:29]: https://github.com/kuutamolabs/near-staking-knd/blob/main/nix/modules/tests/neard.nix

**Comment by yanganto** [comment]
bors merge

**Comment by yanganto** [comment]
bors merge

**Pull request merged**